staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -453,14 +532,31 @@ func (r *crdHandler) createCustomResourceDefinition(obj interface{}) {
 	storageMap := r.customStorage.Load().(crdStorageMap)
 	oldInfo, found := storageMap[crd.UID]
 	if !found {
+		crdInfo, err := r.createServingInfoFor(crd.Name)
+		if err != nil {
+			klog.Errorf("createCustomResourceDefinition failed, error creating new handler: %v", err)
```

Comment:
nit: Previously you mentioned that `createServingInfoFor` may fail if the CRD is just created and it hasn't been approved by the naming controller. When the naming controller approves the name and updates the CRD status,  a new watch event will trigger a successfully handler creation. An optimization here can be detecting whether a naming is approved or not, and logging at a different level (instead of `Errorf`)

Reply:
I think this comment may be obsolete now because I reverted splitting `getOrCreateServingInfoFor()` into 2 parts (as was done previously), and now the only change in `getOrCreateServingInfoFor()` is that it will read from `manager.storageVersionUpdateInfoMap` to know what is the latest CRD. But please correct me if I misunderstood.